rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +611,140 @@ def _query_lending_balances(
 
         return balances
 
+    def query_lending_interests_history(
+            self,
+            cursor: 'DBCursor',
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> bool:
+        """Queries Binance lending interest history, transforms it into `HistoryBaseEntry` objects and saves it in the database.
+
+        "DAILY" -> flexible savings.
+        "CUSTOMIZED_FIXED" -> fixed savings.
+        "ACTIVITY" -> fixed savings but as a result of token partners.
+
+        `end_ts` - `start_ts` <= 30days. This is handled using `_api_query_list_within_time_delta`
+        using `BINANCE_LENDING_INTEREST_HISTORY_INTERVAL_TS` as the timedelta.
+
+        Lending Interest History Documentation:
+        https://binance-docs.github.io/apidocs/spot/en/#get-interest-history-user_data-2
+
+        May raise:
+        - RemoteError
+        - BinancePermissionError
+        """  # noqa: E501
+        ranges = DBQueryRanges(self.db)
+        range_query_name = f'{self.location}_lending_history_{self.name}'
+        ranges_to_query = ranges.get_location_query_ranges(
+            cursor=cursor,
+            location_string=range_query_name,
+            start_ts=start_ts,
+            end_ts=end_ts,
+        )
+        for query_start_ts, query_end_ts in ranges_to_query:
+            events = []
+            for lending_type in ('DAILY', 'ACTIVITY', 'CUSTOMIZED_FIXED'):
+                try:
+                    response = self._api_query_list_within_time_delta(
+                        api_type='sapi',
+                        start_ts=query_start_ts,
+                        time_delta=BINANCE_LENDING_INTEREST_HISTORY_INTERVAL_TS,
+                        end_ts=query_end_ts,
+                        method='lending/union/interestHistory',
+                        additional_options={
+                            'lendingType': lending_type,
+                            'size': BINANCE_LENDING_INTEREST_HISTORY_PAGE_SIZE,
+                        },
+                    )
+                except (RemoteError, BinancePermissionError) as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to query binance lending interest history between '
+                        f'{query_start_ts} and {query_end_ts}. {str(e)}',
+                    )
+                    return True
+
+                for entry in response:
+                    try:
+                        interest_received = deserialize_asset_amount(entry['interest'])  # noqa: E501
+                        if interest_received == ZERO:
+                            continue
+
+                        timestamp = TimestampMS(deserialize_timestamp(entry['time']))
```

Comment:
I don't think this is required again. As far as I can see timestamps in binance are `int` right? So you can just do this, I believe:
```suggestion
                        timestamp = TimestampMS(entry['time'])
```